lib/xclarity_client/configuration.rb

```diff
@@ -1,7 +1,8 @@
 module XClarityClient
   class Configuration
 
-    attr_accessor :username, :password, :host, :auth_type, :port, :verify_ssl, :user_agent_label
+    attr_accessor :username, :password, :host, :auth_type, :port, :verify_ssl,
+                  :user_agent_label, :timeout
 
     def initialize(args)
```

Comment:
I think that is a good opportunity to refactor this method (once the codeclimate is complaining about its complexity). A good start point is split each validation into methods, and just make the calls here.